### ISSUE TITLE
Question about cls loss

### ISSUE THREAD

**Issue by ged0606** [TuSimple/RangeDet#22]
Hi, thanks for you great work.

I implement RangeDet in pytorch, but when i train pedestrian category, the cls loss seems too small, cls loss is two order of magnitude small to reg loss. Does it correct?

Varifocal loss is used as presented in the papar.

```bash
@weighted_loss
def varifocal_loss(pred,
                   target,
                   alpha=1.0,
                   gamma=2.0):
    # pred and target should be of the same size
    assert pred.size() == target.size()

    bce = F.relu(pred) - pred * target
    bce = bce + torch.log(1.0 + torch.exp(-1.0 * pred.abs()))

    positive_mask = (target > 0.0)
    loss_positive = bce * target * positive_mask
    negative_mask = (target <= 0.0)
    pred_score = pred.sigmoid()
    loss_negative = bce * alpha * pred_score.pow(gamma) * negative_mask

    loss = loss_positive + loss_negative

    return loss
```
Looking forward to your reply!

**Comment by ldkong1205** [comment]
Hi @ged0606 , would you like to kindly share your PyTorch code for RangeDet so we can work on this together? I am also interested in this work and want to extend it to semantic segmentation. Thank you very much!

**Comment by ged0606** [comment]
> Hi @ged0606 , would you like to kindly share your PyTorch code for RangeDet so we can work on this together? I am also interested in this work and want to extend it to semantic segmentation. Thank you very much!

The pytorch code is belong to my company, sorry that I can't share out.

**Comment by Abyssaledge** [comment]
I believe it's correct. BTW, you can adopt the Varialfocal Loss in mmdetection, which is equivalent to our implementation and verified effective myself.

**Comment by ldkong1205** [comment]
> > Hi @ged0606 , would you like to kindly share your PyTorch code for RangeDet so we can work on this together? I am also interested in this work and want to extend it to semantic segmentation. Thank you very much!
> 
> The pytorch code is belong to my company, sorry that I can't share out.

Understood 👍

**Comment by ged0606** [comment]
> 

Thanks, I will try it :)
